### ISSUE TITLE
deCONZ: Gateway Firmware Not Connected

### ISSUE THREAD

**Issue by jerrychong25** [marthoc/hassio-addons#84]
Hi,

I have installed the deCONZ hassio addons in Home Assistant of my Raspberry Pi.

However, after installation, firmware version in Phoscon App displays **Not connected**, as per screenshot below:
![jerry pi 1](https://user-images.githubusercontent.com/15714095/53943450-17fc7980-40f8-11e9-94ab-59c3143e70d6.JPG)

Due to this error, I am unable to add new sensors to Phoscon App.

Could anyone advise on this issue?

Thanks.

### deCONZ Config
```
{
  "deconz_device": "/dev/ttyAMA0",
  "web_port": 8080,
  "websockets_port": 8443,
  "debug_info": 1,
  "debug_aps": 0,
  "debug_zcl": 0,
  "debug_zdp": 0,
  "debug_otau": 0,
  "vnc": {
    "active": false,
    "port": 5900,
    "password": "changeme"
  }
}
```

### deCONZ Log
```
[Hass.io] Starting deCONZ Hass.io Addon...
[Hass.io] Current deCONZ version: 2.05.59
[Hass.io] Web UI port: 8080
[Hass.io] Websockets port: 8443
[Hass.io] deCONZ device: /dev/ttyAMA0
[Hass.io] Starting in minimal mode, without VNC GUI
libpng warning: iCCP: known incorrect sRGB profile
This plugin does not support propagateSizeHints()
This plugin does not support propagateSizeHints()
```

**Comment by corvy** [comment]
Did you disable Bluetooth?

**Comment by Rorschach-x** [comment]
I had a real problem for deconz to recognise the Raspbee and spend quite a while on this. The screenshot you provided tells me that the Raspbee has not actually connected, as the firmware is saying not connected. I got it to work in the end with the following steps. I am assuming that you are using a Raspberry Pi 3b+, as the steps before apply to a RPi 3b+.

Can you try changing the ""deconz_device": "/dev/ttyAMA0" to "deconz_device": "/dev/ttyS0" in the deCONZ config.

Also, can you add the following to the end of the config.txt file (and leave a blank line after):

enable_uart=1
dtoverlay=pi3-miniuart-bt
dtoverlay=uart0-full
core_freq=250

**Comment by jerrychong25** [comment]
> Did you disable Bluetooth?

Yes. I disabled Bluetooth.

**Comment by jerrychong25** [comment]
> I had a real problem for deconz to recognise the Raspbee and spend quite a while on this. The screenshot you provided tells me that the Raspbee has not actually connected, as the firmware is saying not connected. I got it to work in the end with the following steps. I am assuming that you are using a Raspberry Pi 3b+, as the steps before apply to a RPi 3b+.
> 
> Can you try changing the ""deconz_device": "/dev/ttyAMA0" to "deconz_device": "/dev/ttyS0" in the deCONZ config.
> 
> Also, can you add the following to the end of the config.txt file (and leave a blank line after):
> 
> enable_uart=1
> dtoverlay=pi3-miniuart-bt
> dtoverlay=uart0-full
> core_freq=250

Let me try on this and will revert back asap.

Thanks.

**Comment by jerrychong25** [comment]
After several trial and errors, only installation method via [deCONZ REST Plugin](https://github.com/dresden-elektronik/deconz-rest-plugin) can displays Firmware Number, and connect to my sensors successfully.

Since sensors are working now, so I will close this issue.

Thanks.

**Comment by marine1988** [comment]
Im facing this error now . Any tip how can I fixe it? im using hassio on a docker in ubuntu server!

**Comment by ars4l4n** [comment]
> add the following to the end of the config.txt file (and leave a blank line after):
enable_uart=1
dtoverlay=pi3-miniuart-bt
dtoverlay=uart0-full
core_freq=250

I did this but it didn' solve the issue.

I then installed deCONZ via the [command line method](https://github.com/dresden-elektronik/deconz-rest-plugin) on Raspbian.
It still doesn't show a firmware number in the deCONZ Hassio Plugin and if I try to connect to deCONZ in the phoscon web app without hassio, my gateway doesn't even show up at all.

What now?
